### PR TITLE
Resolve issue #126

### LINKED ISSUE #126: Importing from cURL
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
Quickly import cURL files to API Dash
Drag & Drop import is preferred.

### PR DESCRIPTION
**Pull request by ketan-sonar**
## PR Description

1. Added "+ Import" button beside "+ New" button.
2. The user can click on it to import a curl file to the app.

## Video of the functionality

https://github.com/foss42/apidash/assets/62693272/2a5177bd-3ea0-4f00-b90e-7a429bcfe517

## Related Issues

- Related Issue #126 
- Closes #126 

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have updated my branch and synced it with project `main` branch before making this PR
- [x] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?

Added tests for httpVerbFromString function and DragAndDropArea widget.

- [x] Yes

### CODE REVIEW

File: lib/utils/convert_utils.dart [hunk 1]
```diff
@@ -179,3 +180,22 @@ List<NameValueModel>? getEnabledRows(
       rows.where((element) => isRowEnabledList[rows.indexOf(element)]).toList();
   return finalRows == [] ? null : finalRows;
 }
+
+HTTPVerb? httpVerbFromString(String value) {
```
Comment (ashitaprasad) [inline lib/utils/convert_utils.dart:184]: This function is not required. Dart has a cleaner way to convert a string to enum by name
https://api.dart.dev/stable/3.4.4/dart-core/EnumByName/byName.html

**Comment by ashitaprasad** [comment]
The code is working, but there are some issues:
- The import functionality is not extensible. It assumes that the input file is of type curl and has not been structured in such a way that it can support other formats
- There is no separation between the UI layer and business logic (curl parsing and converting to request model). I recommend you to take a look at how code generation feature has been implemented.

**Comment by ketan-sonar** [comment]
Okay, I will do the suggested changes.

**Comment by ketan-sonar** [comment]
I have made the suggested changes. Please let me know if there is some issue.

**Comment by ashitaprasad** [comment]
Thanks for taking this initiative @ketan-sonar 🎉

**Comment by ketan-sonar** [comment]
Thank you! I'm glad I could contribute. Looking forward to more opportunities to collaborate and improve the project together.

**Pull request merged**